components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/IdentityCoreServiceComponent.java

```diff
@@ -140,19 +139,7 @@ protected void activate(ComponentContext ctxt) throws MigrationClientException {
 
             String migrate = System.getProperty("migrate");
             String component = System.getProperty("component");
-            try {
-                if (component != null && component.contains("identity") && Boolean.parseBoolean(migrate)) {
-                    log.info("Migration process starting...");
-                    //Directly call migration client here and selectively check for component migrations at client
-                    Class<?> c = Class.forName(MIGRATION_CLIENT_CLASS_NAME);
-                    c.getMethod("databaseMigration").invoke(c.newInstance());
-                    log.info("Migration process finished.");
-                }
-            } catch (Exception e) {
-                log.error("Error occurred while initializing the migration client." , e);
-            }
-
-            if (Boolean.parseBoolean(migrate)) {
+            if (component != null && component.contains("identity") && Boolean.parseBoolean(migrate)) {
```

Comment:
Shall we bring "Boolean.parseBoolean(migrate)" at the front? since we need to check the component only if it is a migrate